### PR TITLE
feat: add contribution documentation

### LINKED ISSUE #4: Provide Code style/contribution documentation
Issue to write up and provide documentation around the code style used and contribution in this codebase. This document should cover the rules, reasons why we enforce them, and acts as a reference for use in PRs if one of the below rules/style choices are not followed. This document  should include (but not be limited to):

- Linting rules applied (using the framework/tools put together in #3)
- Implementation decisions/expectations, including:
  - READMEs being provided for each area of code to explain context/functionality of that area
  - READMEs for all components/as a part of the first contribution
  - TDD, with a focus on user (developer or end user) behaviour/usage (cover the reasons why - more detail to be provided in a Test document)
  - React functional components only using Hooks
  - Context used for state management
  - Storybook stories for all components
  - The `Element/Group/Panel/Bootstrap` component convention (cover the reasons why - more detail to be provided in an architecture document - #6 ) - including the expected files per component type
  - Use of sensible/contextual conventions
- Methodologies to apply (eg BEM, namespacing for styling), and how these are enforced

### PR DESCRIPTION
**Pull request by JQrdan**
Closes: #4

Signed-off-by: Jordan <jordan.tucker1@ibm.com>

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

**Review by matthew-chirgwin — APPROVED** [review]
LGTM! Thanks :) Out of interest re your comment about BEM, are there any stylelint tools which could help check/enforce the format? Might help us reviewers :)

**Comment by JQrdan** [comment]
> LGTM! Thanks :) Out of interest re your comment about BEM, are there any stylelint tools which could help check/enforce the format? Might help us reviewers :)

There's a [BEM plugin](https://www.npmjs.com/package/postcss-bem-linter) for PostCSS if that's something we want to look into using?

**Comment by matthew-chirgwin** [comment]
> > LGTM! Thanks :) Out of interest re your comment about BEM, are there any stylelint tools which could help check/enforce the format? Might help us reviewers :)
> 
> There's a [BEM plugin](https://www.npmjs.com/package/postcss-bem-linter) for PostCSS if that's something we want to look into using?

Rather than bringing in PostCSS, how about https://github.com/namics/stylelint-bem-namics or https://github.com/simonsmith/stylelint-selector-bem-pattern , both of which can be used via our existing stylelint setup?

File: docs/Contribution.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+For full details on what linting is applied, [view our linting documentation](./Linting.md).
+
+## Providing and Updating READMEs
+
+Each area of this codebase has a README that goes alongside it that thouroughly explains the context and functionality of that area. When an area of code is updated, the README should also be updated to reflect the changes. If a new area is being added (e.g. a new panel), that area should also include a new README that explains it. Then, the appropriate related READMEs should be updated to have links to the new documentation.
+
+## Development Method
+
+All components in this repository should be developed using Test Driven Development or Behaviour Driven Development. This ensures that code is only being written that adds value to the end user. By following the TDD or BDD structure properly, it should make refactors safe to do and ensure that regressions can be avoided as much as possible. For more details on writing tests in this repo [view our test documentation](./Test.md).
```
Comment (dlabaj) [inline docs/Contribution.md]: should it say "refactoring" instead of "refactors" ?

File: docs/Contribution.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+## All components should have a Storybook
+
+Every component should also come with a detailed Storybook. They should have appropriate knobs and actions that show the full capability of the component. Developing components with Storybook is a great way to make sure most of this gets done as part of the process, instead of being an afterthough. For more details on Storybook [view our Storybook documentation](./Architecture.md#storybook).
+
+## Component Conventions
+
+We have defined four main types of components that make up the UI: Elements, Groups, Panels, and Bootstrap. These defnitions allow us to clearly split up complexity and allow for maximum modularity of the UI. For full information about why we use this structure [view our architecture documentation](./Architecture.md#element-group-panel-bootstrap-component-pattern)
```
Comment (dlabaj) [inline docs/Contribution.md]: I think we are missing Pages here.

File: docs/Contribution.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+```
+.myComponent{
+  .button-primary {
+
+  }
+  // other child css
+}
+// EOF
+```
```
Comment (dlabaj) [inline docs/Contribution.md:62]: Do we want to Have the CSS in a coding doc that's being added in PR #29 once it's merged?  @matthew-chirgwin @JQrdan

**Review by dlabaj — APPROVED** [review]
LGTM a few minor questions / comments.

File: docs/Contribution.md (hunk 3 above)
Comment (matthew-chirgwin) [reply docs/Contribution.md:62]: I think so - the tooling added will help enforce, but examples are always good too

**Comment by matthew-chirgwin** [comment]
This still LGTM @JQrdan - would you mind addressing the file conflict so we can get this through?

**Pull request merged**